### DISCUSSION TITLE
Allow attaching world effects to a NetId.

### DISCUSSION THREAD

**Discussion by DanielWillett** [SmartlyDressedGames/Unturned-3.x-Community#4414] [General]
Would love to see a way to attach effects to the transform of a NetId, whether its a player, barricade, vehicle, etc.

This could be used for floating icons that apply to a player, vehicle, building, etc. Could also be used to place decals, etc on vehicles or barricades at the exact rotation.

Example:
```cs
TriggerEffectParameters p = new TriggerEffectParameters(asset)
{
   Transform = player.GetNetId()
};

p.SetRelevantPlayer(otherPlayer.channel.owner);

EffectManager.triggerEffect(p);
```

**Comment by SDGNelson** [comment]
My first thought was that I thought I added this already. It is supported for certain effects like blood splatters and bullet holes, but wasn't further expanded yet because it relies on the attachment parent calling `EffectManager.ClearAttachments` before destroy - otherwise `EffectManager.attachedEffects` leaks. :[
